python/src/lib.rs

```diff
@@ -113,7 +115,54 @@ fn register_datagen(py: Python, m: &Bound<'_, PyModule>) -> PyResult<()> {
 }
 
 // TODO: make this runtime configurable (e.g. num threads)
-static RT: LazyLock<BackgroundExecutor> = LazyLock::new(BackgroundExecutor::new);
+static RT: atomic::AtomicPtr<BackgroundExecutor> = atomic::AtomicPtr::new(std::ptr::null_mut());
+
+/// Get the current global executor object.
+pub fn global_executor() -> &'static mut BackgroundExecutor {
+    let ptr = RT.load(Ordering::SeqCst);
+    if !ptr.is_null() {
+        return unsafe { &mut *ptr }
+    }
+    let new_ptr = Box::into_raw(Box::new(BackgroundExecutor::new()));
+    match RT.compare_exchange(
+        std::ptr::null_mut(),
+        new_ptr,
+        Ordering::SeqCst,
+        Ordering::SeqCst,
+    ) {
+        Ok(_) => {
+            unsafe { &mut *new_ptr }
+        },
+        Err(racing_ptr) => unsafe {
+            // Another thread already installed an executor object. Drop our new executor, which
+            // is no longer needed.
+            drop(Box::from_raw(new_ptr));
+            &mut *racing_ptr
+        },
+    }
+}
+
+/// Handle the process forking.
+///
+/// After a fork() operation, threads from the parent process do not carry over to the child.
+/// The child process needs to re-create its BackgroundExecutor object.
+///
+/// atfork handlers run in "async-signal context" which means that we can't (safely) do much here.
+/// Therefore, we just set the global executor pointer to null, and let the next thread that wants
+/// to use the global executor re-create it.
+///
+/// Note also that this function does leak memory, because it doesn't clean up the old executor
+/// pointer. This is intentional.
+extern "C" fn atfork_child() {
+    RT.store(std::ptr::null_mut(), Ordering::SeqCst);
+}
+
+fn install_atfork() {
+    let result = unsafe {
+        libc::pthread_atfork(None, None, Some(atfork_child))
+    };
+    assert_eq!(result, 0, "pthread_atfork failed");
+}
```

Comment:
Might need to wrap this for Windows?

Reply:
Yeah, good point. I will stub this out on Windows. Windows has its own special threading system which doesn't conform to pthreads. It's not entirely clear what `pthread_atfork` does there, but from what I've read it's nothing useful.